mitodl/micromasters
# add basic HTML email capability

#### What are the relevant tickets?

closes #2948 

#### What's this PR do?

This adds support for basic HTML rich text to our emails. For rich-text editing we're using [react-draft-wysiwyg](https://github.com/jpuri/react-draft-wysiwyg), which is a slim wrapper around Facebook's [draft-js](https://github.com/facebook/draft-js). It's pretty slick!
 
#### How should this be manually tested?

Here are things to test:

- check that you can use HTML to compose an email, and that the HTML is displayed accurately in the email you receive
- check that a reasonable text fallback is generated. it should basically have all the text in the email, and if there are any anchor tags in the HTML it should replace those anchor tags with their `href` attributes.
- when you're composing an email you should get a validation message if you add a link to the email whose `href` attribute does not start with `http|https` (this relates to a restriction that draft-js enforces - if you try to instantiate an `EditorState` object, which is uses to manage its internal state, with an HTML string containing anchor tags which don't meet this requirement it will just turn those anchor tags into plain text - not what we want).
- if you save an HTML `AutomaticEmail` (from `/learners`) you should be able to edit the HTML of that `AutomaticEmail` on the automatic email admin page without any problems.

that's all I can think of specifically...

![email_editing](https://cloud.githubusercontent.com/assets/6207644/25195011/50a3bade-250a-11e7-9b33-395af3aada9c.png)
